contracts/interchain-token-service/InterchainTokenService.sol

```diff
@@ -903,4 +840,36 @@ contract InterchainTokenService is
             amount
         );
     }
+
+    /**
+     * @notice Transmit a sendTokenWithData for the given tokenId. Only callable by a token manager.
+     * @param tokenId the tokenId of the TokenManager (which must be the msg.sender).
+     * @param sourceAddress the address where the token is coming from, which will also be used for reimburment of gas.
+     * @param destinationChain the name of the chain to send tokens to.
+     * @param destinationAddress the destinationAddress for the interchainTransfer.
+     * @param amount the amount of token to give.
+     * @param metadata the data to be passed to the destiantion.
+     */
+    function _transmitSendToken(
+        bytes32 tokenId,
+        address sourceAddress,
+        string calldata destinationChain,
+        bytes memory destinationAddress,
+        uint256 amount,
+        bytes memory metadata
+    ) internal {
+        bytes memory payload;
+        if (metadata.length < 4) {
+            payload = abi.encode(SELECTOR_SEND_TOKEN, tokenId, destinationAddress, amount);
```

Comment:
still thinking that we should combine the two and take the gas penalty since the code is much simpler